feat: support with ties in FetchRel (#797)
substrait-io/substrait: site/docs/relations/logical_relations.md

## Fetch Operation

> The fetch operation eliminates records outside a desired window. Typically corresponds to an fetch/offset SQL clause. Will only returns records between the start offset and the end offset.

Reviewer: I think `a` is correct here?

Reviewer: I tried asking Claude and it gave me both answers :laughing:

![image](https://github.com/user-attachments/assets/3f3db107-0ffc-49ec-9dca-122c8136c0a5)

Author: Haha! I typed `an ORDER BY...` then removed `ORDER BY` but left `n` there. :) fixed.

| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------- |
| Input             | A relational input.                                                                                                                                                                                                                  | Required                   |
| Offset Expression | An expression which evaluates to a non-negative integer or null (recommended type is `i64`). Declares the offset for retrieval of records. An expression evaluating to null is treated as 0.                                         | Optional, defaults to a 0 literal.   |
| Count Expression  | An expression which evaluates to a non-negative integer or null (recommended type is `i64`). Declares the number of records that should be returned. An expression evaluating to null indicates that all records should be returned. | Optional, defaults to a null literal. |

> | Sort Fields       | List of one or more sort fields to define a desired window. Requires at least one sort field if 'With Ties' is true.                                                                                                                 | Optional, defaults to empty sort fields. |

Reviewer: ```suggestion
| Sort Fields       | List of one or more sort fields to define a desired orderedness. Requires at least one sort field if 'With Ties' is true.                                                                                                                 | Optional, the default will preserve the input orderedness |
```

> | With Ties         | Whether to return tie rows of the last row specified by 'Count Expression'. If false, at most 'Count Expression' records are returned. If true, it may yield more recrods than 'Count Expression'.                                   | Optional, defaults to false. |

Reviewer: ```suggestion
| With Ties         | Whether to return "tied rows" which are rows equal to last row that would be returned by 'Count Expression'. If false, at most 'Count Expression' records are returned. If true, it may yield more records than 'Count Expression'.                                   | Optional, defaults to false. |
```
